plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/server/debug/OpenShiftDebugUtils.java

```diff
@@ -0,0 +1,260 @@
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.tools.openshift.common.core.OpenShiftCoreException;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
+import org.jboss.tools.openshift.internal.core.models.PortSpecAdapter;
+import org.jboss.tools.openshift.internal.core.util.ResourceUtils;
+
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.resources.IClientCapability;
+import com.openshift.restclient.model.IContainer;
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.IEnvironmentVariable;
+import com.openshift.restclient.model.IPod;
+import com.openshift.restclient.model.IPort;
+
+public class OpenShiftDebugUtils {
+
+	private ILaunchManager launchManager;
+
+	public static OpenShiftDebugUtils get() {
+		return get(DebugPlugin.getDefault().getLaunchManager());
+	}
+	
+	/** For testing purposes **/
+	public static OpenShiftDebugUtils get(ILaunchManager launchManager) {
+		return new OpenShiftDebugUtils(launchManager);
+	}
+	
+	private OpenShiftDebugUtils(ILaunchManager launchManager) {
+		this.launchManager = launchManager;
+	}
+	
+	public DebuggingContext enableDebugMode(IDeploymentConfig deploymentConfig, DebuggingContext debugContext, IProgressMonitor monitor) throws CoreException {
+		assert deploymentConfig != null;
+		assert debugContext != null;
+
+		IDebugListener listener = debugContext.getDebugListener();
+		if (debugContext.isDebugEnabled() && listener != null) {
+			IPod pod = getFirstPod(deploymentConfig);
+			debugContext.setPod(pod);
+			listener.onDebugChange(debugContext, monitor);
+		} else {
+			debugContext.setDebugEnabled(true);
+			updateDebugConfig(deploymentConfig, debugContext, monitor);
+		}
+		return debugContext;
+	}
+
+	public DebuggingContext disableDebugMode(IDeploymentConfig deploymentConfig, DebuggingContext debugContext, IProgressMonitor monitor) throws CoreException {
+		assert deploymentConfig != null;
+		assert debugContext != null;
+
+		IDebugListener listener = debugContext.getDebugListener();
+		if (listener != null) {
+			IPod pod = getFirstPod(deploymentConfig);
+			debugContext.setPod(pod);
+			listener.onDebugChange(debugContext, monitor);
+		} 
+		if (debugContext.isDebugEnabled()) {
+			debugContext.setDebugEnabled(false);
+			updateDebugConfig(deploymentConfig, debugContext, monitor);
+		}
+		return debugContext;
+	}
+	
+	public void updateDebugConfig(IDeploymentConfig deploymentConfig, DebuggingContext debugContext, IProgressMonitor monitor) throws CoreException {
+		monitor.subTask("Updating Deployment Configuration");
+		if (deploymentConfig == null
+				|| deploymentConfig.getEnvironmentVariables() == null) {
```

Comment:
I dont think this will ever return null for ENV vars, but its worth keeping the check.  What about a check or empty too?